States/graph.c

```diff
@@ -0,0 +1,236 @@
+#include <stdlib.h>
+#include "graph.h"
+#include <stdio.h>
+#include <math.h>
+#include <stdbool.h>
+#include <string.h>
+
+typedef struct Road {
+    int start;
+    int end;
+    int length;
+} Road;
+
+typedef struct State {
+    int number;
+    int* cities;
+    int citiesCount;
+} State;
+
+void updateDistances(int city, int citiesCount, int* distances, int** adjacencyMatrix, bool* alreadyDistributedCities) {
+    for (int i = 0; i < citiesCount; i++) {
+        if (alreadyDistributedCities[i] != true) {
```

Comment:
Стайлгайд запрещает сравниваться с булевыми константами (потому что это тавтология), правильно так:
```suggestion
        if (!alreadyDistributedCities[i]) {
```